validator/embedded/src/main/java/org/ethereum/beacon/validator/BeaconChainAttester.java

```diff
@@ -0,0 +1,78 @@
+package org.ethereum.beacon.validator;
+
+import org.ethereum.beacon.chain.observer.ObservableBeaconState;
+import org.ethereum.beacon.consensus.BeaconChainSpec;
+import org.ethereum.beacon.core.BeaconState;
+import org.ethereum.beacon.core.operations.Attestation;
+import org.ethereum.beacon.core.operations.attestation.AttestationData;
+import org.ethereum.beacon.core.operations.attestation.AttestationDataAndCustodyBit;
+import org.ethereum.beacon.core.types.BLSSignature;
+import org.ethereum.beacon.core.types.ShardNumber;
+import org.ethereum.beacon.core.types.SlotNumber;
+import org.ethereum.beacon.core.types.ValidatorIndex;
+import org.ethereum.beacon.validator.crypto.MessageSigner;
+import tech.pegasys.artemis.ethereum.core.Hash32;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+import static org.ethereum.beacon.core.spec.SignatureDomains.ATTESTATION;
+
+/**
+ * An interface of beacon chain attester. A part of beacon validator logic.
+ *
+ * @see ValidatorService
+ */
+public interface BeaconChainAttester {
+
+  /**
+   * Creates {@link AttestationDataAndCustodyBit} instance and signs off on it.
+   *
+   * @param state a state at a slot that validator is attesting in.
+   * @param data an attestation data instance.
+   * @param signer message signer.
+   * @param spec Beacon chain spec
+   * @return signature of attestation data and custody bit.
+   */
+  static BLSSignature getAggregateSignature(
```

Comment:
Why so static? :) I mean, why this code should be in a static method declared in top level interface?